docs/testing.md

```diff
@@ -0,0 +1,34 @@
+# Testing of Custom Type Errors
+
+Ema uses static verification to catch most common errors related to anyclass deriving of `GenericRoute`, instead displaying to the user more legible and friendly error messages. Due to current GHC limitations, we are left with having to _manually_ test that custom type errors are, in fact, displaying appropriately for various cases of improper deriving. That said, the current testing methodology has been designed to ease the transition over to future automated testing as GHC limitations are eventually resolved.
+
+## The Problem
+
+- Automated testing is ultimately being held up on the now-implemented `Unsatisfiable` proposal landing in a release GHC version.
+  * Due to more formalized semantics, this will allow for custom type errors that can be delayed until runtime via `-fdefer-type-errors`
+  * Relevant PR documenting the fact that `TypeError`s do not actually defer even with `-fdefer-type-errors` can be found [here](https://github.com/CRogers/should-not-typecheck/pull/15#issuecomment-502295932)
+
+## Current Manual Workaround
+
+Current release candidates of Ema should load a REPL for the `test-type-errors` test suite (ex. `nix develop -c cabal -- repl test-type-errors`), and for each test case:
+
+1. Comment out the `#undef` at the top of each test case.
+2. Reload the source file (`:r`).
```

Comment:
I don't plan to use GHCi, but I do use HLS. When I comment out the first `#undef` while editing the file in VSCode I see something unexpected ... parse error, which is strange:

<img width="265" alt="image" src="https://user-images.githubusercontent.com/3998/181258741-a222c076-aa67-48a5-986c-e2f51a6f6f6a.png">
 
ghcid in terminal reports:

Hovering over:

<img width="343" alt="image" src="https://user-images.githubusercontent.com/3998/181258891-7f980e76-163c-470b-b66d-cf82316a8fec.png">

<img width="335" alt="image" src="https://user-images.githubusercontent.com/3998/181258788-6f6e99b4-3240-4bc0-ab4c-51ced189fd44.png">